File: src/main/java/net/thedragonteam/armorplus/client/gui/base/GuiBaseBench.java
package net.thedragonteam.armorplus.client.gui.base;

import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.inventory.Container;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.text.TextComponentTranslation;

import static java.lang.String.format;
import static net.thedragonteam.armorplus.util.Utils.setRL;

public class GuiBaseBench extends GuiBase {

    private final int color;
    private ResourceLocation resourceLocation;
    private String name;

    public GuiBaseBench(Container container, ResourceLocation resourceLocation, String benchName, int xSize, int ySize, int color) {
        super(container);
        this.resourceLocation = resourceLocation;
        this.name = benchName;
        this.xSize = xSize;
        this.ySize = ySize;
        this.color = color;
    }

    public GuiBaseBench(Container container, String benchName, int xSize, int ySize, int color) {
        this(container, setRL(format("textures/gui/container/gui_%s.png", benchName)), benchName, xSize, ySize, color);
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        this.drawDefaultBackground();
        super.drawScreen(mouseX, mouseY, partialTicks);
        this.renderHoveredToolTip(mouseX, mouseY);
    }

    /**
     * Draw the foreground layer for the GuiContainer (everything in front of the items)
     */
    @Override
    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
        this.fontRenderer.drawString(new TextComponentTranslation("container.armorplus." + name).getFormattedText(), 28, 5, color);
        this.fontRenderer.drawString(new TextComponentTranslation("container.armorplus.inventory").getFormattedText(), 8, this.ySize - 96 + 2, color);
    }

    /**
     * Draws the background layer of this container (behind the items).
     */
    @Override
    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
        GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
        this.mc.getTextureManager().bindTexture(resourceLocation);
        int i = (this.width - this.xSize) / 2;
        int j = (this.height - this.ySize) / 2;
        this.drawTexturedModalRect(i, j, 0, 0, this.xSize, this.ySize);
    }
}

File: src/main/java/net/thedragonteam/armorplus/client/gui/base/GuiBaseBench.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.client.gui.base

import net.minecraft.client.gui.inventory.GuiContainer
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.inventory.Container
import net.minecraft.util.ResourceLocation
import net.thedragonteam.armorplus.util.TextUtils

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
open class GuiBaseBench(container: Container, var resourceLocation: ResourceLocation, var name: String, xSize: Int, ySize: Int) : GuiContainer(container) {

    init {
        this.xSize = xSize
        this.ySize = ySize
    }

    /**
     * Draw the foreground layer for the GuiContainer (everything in front of the items)
     */
    override fun drawGuiContainerForegroundLayer(mouseX: Int, mouseY: Int) {
        this.fontRenderer.drawString(TextUtils.formattedText("container.armorplus." + name), 28, 5, 4210752)
        this.fontRenderer.drawString(TextUtils.formattedText("container.armorplus.inventory"), 8, this.ySize - 96 + 2, 4210752)
    }

    /**
     * Draws the background layer of this container (behind the items).
     */
    override fun drawGuiContainerBackgroundLayer(partialTicks: Float, mouseX: Int, mouseY: Int) {
        GlStateManager.color(1.0f, 1.0f, 1.0f, 1.0f)
        this.mc.textureManager.bindTexture(resourceLocation)
        val i = (this.width - this.xSize) / 2
        val j = (this.height - this.ySize) / 2
        this.drawTexturedModalRect(i, j, 0, 0, this.xSize, this.ySize)
    }
}
